views/js/runner/plugins/controls/progressbar/progress.js

```diff
@@ -252,12 +307,36 @@ define([
                 if (section.stats.questions > 0) {
                     updateDetails(stats.answerableSections, section, currentItem.position);
                 }
+
+                if (config.indicator === 'categories') {
+                    _.forEach(section.items, function (item) {
+                        if (matchCategories(item.categories, categoriesToMatch)) {
+                            updateItemDetails(stats.matchedCategories, item, currentItem.position);
+                        }
+                    });
+                }
             });
         });
 
         return stats;
     }
 
+    function matchCategories(categories, expectedCategories) {
+        var categoryId;
+
+        if (expectedCategories && expectedCategories.length) {
+            // all expected categories have to be included in the provided categories
+            for (categoryId in expectedCategories) {
```

Comment:
Since `expectedCategories` is an `array`, you should not loop on it using something reserved to `object`...
It could work, but you will go into troubles